[TASK] Add missing directories (#2256)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ApiOverview/DirectoryStructure/Index.rst

Also see :ref:`Environment` for further information, especially how to retrieve
the paths within PHP code.

Files on project level
======================

On the top most level, the project level, you can find the files
:file:`composer.json` which contains requirements for the TYPO3 installation
and the :file:`composer.lock` which contains information about the concretely
installed versions of each package.

Directories in a typical project
================================

.. contents::
   :local:

.. _directory-config:

:file:`config/`
---------------

> Composer installations only. TYPO3 configuration directory. This directory

Reviewer: ```suggestion
Composer-based installations only. TYPO3 configuration directory. This directory
```

contains installation-wide configuration.

.. _directory-config-sites:

:file:`config/sites/`
~~~~~~~~~~~~~~~~~~~~~

> Composer installations only.  The folder :file:`config/sites` contains

Reviewer: ```suggestion
Composer-based installations only.  The folder :file:`config/sites` contains
```

subfolders for each :ref:`site configuration <sitehandling>`.

.. _directory-local_packages:

> :file:`local_packages/`

Reviewer: When you introduce now a "local_packages" folder, please take care of other places in the TYPO3 docs, where "packages" is used (to be consistent), for example here: https://docs.typo3.org/m/typo3/guide-installation/main/en-us/MigrateToComposer/BestPractices.html#define-a-local-path-repository
I assume, there are other places.

Author: We already use local_packages in the sitepackage and would take care of this in a follow up

-----------------------

> Composer installations only.  Each web site which is run on TYPO3 **should**

Reviewer: ```suggestion
Composer-based installations only.  Each web site which is run on TYPO3 **should**
```

> have a sitepackage, anextension with a special purpose containing all

Reviewer: ```suggestion
have a sitepackage, an extension with a special purpose containing all
```

templates, styles, images, etc. needed for the theme.

> It is usually stored locally and then symlinked into the :ref:`directory-vendor`

Reviewer: That is not true for v11 with composer-installers v3 (the majority) - and as this is merged into v11 it should be `typo3conf/ext`.

Author: Will be dealed with in a follow up

folder. Many projects also need custom extensions that can be stored here.

The folder for local packages has to be defined in the project's :file:`composer.json`
to be used:

..  code-block:: json

>     :caption: composer.json

Reviewer: ```suggestion
    :caption: composer.json

```

The snippet is not rendered, probably due to missing empty line.

    {
        "name": "myvendor/my-project",
        "repositories": {
            "my_local_packages": {
                "type": "path",
                "url": "local_packages/*"
            }
        },
        "...": "..."
    }

.. _directory-public:

:file:`public/`
---------------

We assume here that your web root points to a folder called :file:`public` in
a Composer-based installation or a legacy installation as is commonly done.
Otherwise, replace  :file:`public` with the path to your web root. In legacy

> installations by default all files are located in this folder.

Reviewer: "In legacy installations, all files are located in this folder by default."

Author: will be solved in a follow up

> installations by default all files are located in this folder.

Author: ```suggestion
installations all files are located in this folder by default..
```

Also see :ref:`Environment` for further information, especially how to retrieve
the paths within PHP code.

Files on project level
======================

On the top most level, the project level, you can find the files
:file:`composer.json` which contains requirements for the TYPO3 installation

> and the :file:`composer.lock` which contains information about the concretely

Reviewer: ```suggestion
and the :file:`composer.lock` which contains information about the concrete
```

:file:`AdditionalConfiguration.php`. See chapter
:ref:`Configuration files <configuration-files>` for details.

.. _directory-public-typo3conf-autoload:

:file:`public/typo3conf/autoload/`
""""""""""""""""""""""""""""""""""

Legacy installations only. Contains :ref:`autoloading <autoload>` information.

> The files are update each time an extension is installed in the

Reviewer: ```suggestion
The files are updated each time an extension is installed via the
```

Directory for local TYPO3 extensions. Each subdirectory contains one extension.

.. _directory-public-typo3conf-l10n:

:file:`public/typo3conf/l10n/`
""""""""""""""""""""""""""""""

> Legacy installations only. Directory for extension localisations.

Reviewer: ```suggestion
Legacy installations only. Directory for extension localizations.
```

As we use American English in docs.

Contains all downloaded translation files.

.. _directory-public-typo3conf-sites:

:file:`public/typo3conf/sites/`
"""""""""""""""""""""""""""""""

> Legacy installations only. Contains contains

Reviewer: ```suggestion
Legacy installations only. Contains
```

Directory for temporary files that should be public available
(e.g. generated images).

.. _directory-public-typo3temp-var:

:file:`public/typo3temp/var/`
"""""""""""""""""""""""""""""

Legacy installations only. Directory for temporary files that contains private

> files (e.g. cached Fluid templates) and should not be publicly available.

Reviewer: ```suggestion
files (e.g. cache and log files) and should not be publicly available.
```

See also :ref:`Environment-configuring-paths` for a more detailed description.

.. _directory-var:

:file:`var/`
------------

> Composer installations only. Directory for temporary files that contains private files (e.g.

Reviewer: ```suggestion
Composer-based installations only. Directory for temporary files that contains private files (e.g.
```

cache and logs files) and should not be publicly available.
See also :ref:`Environment-configuring-paths` for a more detailed description.

.. _directory-var-cache:

:file:`var/cache/`
~~~~~~~~~~~~~~~~~~~

> Composer installations only. This directory contains internal files needed

Reviewer: ```suggestion
Composer-based installations only. This directory contains internal files needed
```
Hmm, not really "needed for the cache". If the files are not there, they are generated. Perhaps better: ".. contains cache files".

for the cache.

.. _directory-var-labels:

:file:`var/labels/`
~~~~~~~~~~~~~~~~~~~

> Composer installations only.  The directory :file:`var/labels/` is for extension

Reviewer: ```suggestion
Composer-based installations only.  The directory :file:`var/labels/` is for extension
```

"... is for localizations. It contains all downloaded translation files." -> maybe better: "... contains all downloaded translations files." Just omit the "is for" part.

> localisations. It contains all downloaded translation files.

Reviewer: ```suggestion
localizations. It contains all downloaded translation files.
```

This path can be retrieved from the Environment API, see
:ref:`Environment-labels-path`.

.. _directory-var-log:

:file:`var/log/`
~~~~~~~~~~~~~~~~~~~

> Composer installations only.  This directory contains log files like the

Reviewer: ```suggestion
Composer-based installations only.  This directory contains log files like the
```

TYPO3 log, the deprecations log and logs generated by extensions.

.. _directory-vendor:

:file:`vendor/`
---------------

> Composer installations only.  In this directory, which lies outside of

Reviewer: ```suggestion
Composer-based installations only.  In this directory, which lies outside of
```

the webroot, all extensions (system, third-party and custom) are installed
as Composer packages.

The directory contains folders for each required vendor and inside each
vendor directory there is a folder with the different project names.

For example the system extension `core` has the complete package name
`typo3/cms-core` and will therefore be installed into the directory
:file:`vendor/typo3/cms-core`. The extension `news`, package name
`georgringer/news` will be installed into the folder
:file:`vendor/georgringer/news`.

> Never put or symlink your extensions manually into this directory as they would

Reviewer: Hmm, would rephrase it: "... as it is managed by Composer and any manual changes are getting lost, e.g. on deployment".

> Never put or symlink your extensions manually into this directory as they would
> not be found. Local extensions and sitepackages should be kept in a separate
> folder outside the web root, for example

Author: ```suggestion
Never put or symlink your extensions manually into this directory as it is 
managed by Composer and any manual changes are getting lost, 
for example on deployment. Local extensions and sitepackages 
should be kept in a separate folder outside the web root, for example
```

:ref:`local_packages <directory-local_packages>`.

> Upon installation , Composer creates a symlink from local_packages to

Reviewer: As mentioned above, only true for cms-composer-installers v4+. And mainly not TYPO3 v11.5.

Author: will be done in a follow up
